Crash when running appimagetool in git repository without a single commit

```bash
❯ ../appimagetool-551-x86_64.AppImage AppDir

appimagetool-551-x86_64.AppImage 551

2020/08/25 19:36:06 git root: /home/srevinsaju/repo/scrcpy-test
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x83a31a]

goroutine 1 [running]:
main.GenerateAppImage(0x7ffe9a8d9e2d, 0x6)
        github.com/probonopd/go-appimage/src/appimagetool/appimagetool.go:264 +0x44da
main.main()
        github.com/probonopd/go-appimage/src/appimagetool/appimagetool.go:217 +0x912
```
~~Well,I do not know what happened~~ See https://github.com/probonopd/go-appimage/issues/62#issuecomment-680141958 .

Confirmed.

Without `git init`, the error does not happen:

```
me@host:~/test$ ./appimagetool-551-x86_64.AppImage AppDir/

appimagetool-551-x86_64.AppImage 551

2020/08/25 20:39:35 Apparently not in a git repository
Version not found, aborting. Set it with VERSION=... ./appimagetool-551-x86_64.AppImage
```

With git init but no commits the error does happen:

```
me@host:~/test$ git init
Initialized empty Git repository in /home/me/test/.git/
me@host:~/test$ ./appimagetool-551-x86_64.AppImage AppDir/

appimagetool-551-x86_64.AppImage 551

2020/08/25 20:39:41 git root: /home/me/test
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x83a31a]

goroutine 1 [running]:
main.GenerateAppImage(0x7ffca34733ff, 0x6)
	github.com/probonopd/go-appimage/src/appimagetool/appimagetool.go:264 +0x44da
main.main()
	github.com/probonopd/go-appimage/src/appimagetool/appimagetool.go:217 +0x912
```

Committing something makes the error go away:

```
me@host:~/test$ git add AppDir/
me@host:~/test$ git commit -m "Initial commit"
[master (root-commit) d1b1504] Initial commit
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 AppDir/AppRun
me@host:~/test$ ./appimagetool-551-x86_64.AppImage AppDir/

appimagetool-551-x86_64.AppImage 551

2020/08/25 20:40:20 git root: /home/me/test
2020/08/25 20:40:20 NOTE: Using d1b1504 from 'git rev-parse --short HEAD' as the version
2020/08/25 20:40:20       Please set the $VERSION environment variable if this is not intended
No top-level desktop file found in AppDir, aborting
```

--- a/src/appimagetool/appimagetool.go
+++ b/src/appimagetool/appimagetool.go
@@ -260,11 +260,11 @@ func GenerateAppImage(appdir string) {
 			log.Println("git root:", gitRoot)
 			if version == "" {
 				gitHead, err := gitRepo.Head()
-				version = gitHead.Hash().String()[:7] // This equals 'git rev-parse --short HEAD'
 				if err != nil {
 					os.Stderr.WriteString("Could not determine version automatically, please supply the application version as $VERSION " + filepath.Base(os.Args[0]) + " ... \n")
 					os.Exit(1)
 				} else {
+					version = gitHead.Hash().String()[:7] // This equals 'git rev-parse --short HEAD'
 					log.Println("NOTE: Using", version, "from 'git rev-parse --short HEAD' as the version")
 					log.Println("      Please set the $VERSION environment variable if this is not intended")
 				}

